### PR TITLE
All together now

### PR DESCRIPTION
**Pull request by bukzor**
This is my other four PRs merged up.

There were a couple conflicts.

### CODE REVIEW

File: src/clom/command.py [hunk 1]
```diff
@@ -22,6 +9,17 @@ def inner(*args, **kwargs):
     'OR',
 ]
 
+
+def decorator(decoration):
+    def new_decorator(wrapped):
+        from functools import wraps
```
Comment (six8) [inline src/clom/command.py]: Move import to top

File: src/clom/command.py [hunk 2]
```diff
@@ -293,8 +294,10 @@ def _clone(self):
     def __eq__(self, other):
         if isinstance(other, string_types):
             return other == str(self)
+        elif isinstance(other, Operation):
+            return vars(other) == vars(self)
         else:
-            return super(self.__class__, self).__eq__(other)
+            return NotImplemented
```
Comment (six8) [inline src/clom/command.py:301]: This should be `raise NotImplemented()`, but really I think we should just `return False` and say they're not equal.

File: src/clom/command.py [hunk 3]
```diff
@@ -354,13 +357,22 @@ def __init__(self, clom, name, parent=None):
...
+                self._kwopts[name] = val
+
     @_makes_clone
     def with_opts(self, *args, **kwargs):
-        """
+        r"""
         Options to call the command with.
 
         :param kwargs: A dictionary of options to pass to the command.
                        Keys are generated as `--name value` or `-n value` depending on the length.
```
Comment (six8) [inline src/clom/command.py]: Example should be `--name=value` now.

File: src/clom/command.py (hunk 2 above)
Comment (bukzor) [inline src/clom/command.py:301]: Respectfully, you are mistaken. https://docs.python.org/2/library/constants.html#NotImplemented

Comment (six8) [inline src/clom/command.py:301]: Ah, thanks, never came across that one. Would this result in an exception or just result in False for the comparison?

Comment (bukzor) [inline src/clom/command.py:301]: It results in graceful degredation wherein the `other` may implement `__eq__`, and failing that, we get False.

File: src/clom/command.py (hunk 3 above)
Comment (bukzor) [inline src/clom/command.py]: I did a quick grep, and I think this is the only one I missed.

File: src/clom/command.py [hunk 4]
```diff
@@ -354,13 +358,22 @@ def __init__(self, clom, name, parent=None):
         # Arguments to pass to the command line
         self._args = []
 
+    def __sort_kwargs(self, kwargs):
```
Comment (six8) [inline src/clom/command.py]: Agreed that this is a bit awkward but it works. Maybe naming it `_parse_kwargs`?

Comment (bukzor) [inline src/clom/command.py]: Dealer's choice. Feel free to fixup after merge.

**Comment by six8** [comment]
Looks good but should target develop. I merged master into develop so it should be in correct state now. Also need to bump version.

**Comment by bukzor** [comment]
Can all be done more easily post-merge.

**Comment by bukzor** [comment]
If you want pull requests to be made versus `develop` branch, there's a setting for that https://help.github.com/articles/setting-the-default-branch/

**Comment by six8** [comment]
Thanks for reminding me. Guess I never got around to setting that on this project :).

**Comment by six8** [comment]
Merged into develop via #20

**Pull request closed without merging**